### ISSUE TITLE
Companion.insert useless when creating database from .moor file

### ISSUE THREAD

**Issue by knaeckeKami** [simolus3/moor#392]
db.moor:

```sql
CREATE TABLE FROMMOOR (
  "textColumn" text
)
```

from_moor.dart:

```dart
import 'dart:io';

import 'package:moor/moor.dart';
import 'package:moor_ffi/database.dart';
import 'package:moor_ffi/moor_ffi.dart';

part 'from_moor.g.dart';


@UseMoor(include: {"db.moor"})
class FromMoorDb extends _$FromMoorDb {
  FromMoorDb.open(File file) : super(VmDatabase(file));

  int get schemaVersion => 1;
}
```

Generates: 

```dart
  FrommoorCompanion.insert({
    this.textColumn = const Value.absent(),
  });
```

Whereas:

```dart
import 'dart:io';

import 'package:moor/moor.dart';
import 'package:moor_ffi/database.dart';
import 'package:moor_ffi/moor_ffi.dart';

part 'from_dart.g.dart';

@UseMoor(tables: [FromDart])
class FromDartDb extends _$FromDartDb {
  FromDartDb.open(File file) : super(VmDatabase(file));

  int get schemaVersion => 1;
}

class FromDart extends Table {
  TextColumn get textColumn => text()();
}

```

Generates:

```dart
  FromDartCompanion.insert({
    @required String textColumn,
  }) : textColumn = Value(textColumn);
```

the `FrommoorCompanion.insert` is pretty useless, as it is identical to the unnamed constructor:

```dart
 const FrommoorCompanion({
    this.textColumn = const Value.absent(),
  });
```

I think that's a bug, isn't it? Manually wrapping values into Value<T> objects is quite cumbersome.

**Comment by simolus3** [comment]
TL;DR: Columns from Dart tables are non-nullable by default, sqlite columns aren't.

In your example, `textcolumn` doesn't have a `NOT NULL` constraint, so it accepts null as an value. In fact, null would be the default value during an insert. So, moor decides that this column doesn't have to be set and wraps it in a `Value` so that it can be `Value.absent()`.
On the other hand, Dart columns generate a `NOT NULL` constraint by default, unless one uses the `nullable()` method. Obviously, attempting to insert a null value (which would be the default) into a `NOT NULLABLE` column doesn't make much sense. Since `Value.absent()` isn't really an allowed state here, moor expects the unwrapped type instead and puts an `@required` onto it as well. This is only happening on the `.insert` constructor because omitting that value on updates would be perfectly legal - you'd just leave the old value.

I'd say that moor is working as expected here, although it can be a little confusing. Perhaps the documentation should be improved to point out that Dart columns are NNBD whereas columns from a `CREATE TABLE` statement aren't.

Maybe we should also omit the `insert` constructor when it would be identical to the regular constructor (or just make it an redirecting factory constructor).

**Comment by knaeckeKami** [comment]
Ah, makes sense. I didn't think about nullability.
